### PR TITLE
Makes all global variables handled by the GLOB controller

### PR DESCRIPTION
**Pull request by AffectedArc07**
# NOTE
Unlike the last PR, this shouldnt merge-conflict every PR in existence. While it touches a lot of files, 99% of the changes are just adding in `GLOB.` to variables.

## What Does This PR Do
This PR makes all global variables in the code managed globals, and adds a CI step to ensure future globals are made using the appropriate handlers.

It also fixes some random rogue var delcarations, and a few bugs (Such as buildmode overlays not showing up for some things)

## Why It's Good For The Game
Our current mess of global variables did not allow for easy debugging, or even knowing how many we have. This PR fixes that, and also makes it so the same mistakes cant be repeated.

## Changelog
:cl: AffectedArc07
tweak: All global variables are now managed
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
This will make pinning down GC issues a lot easier too.

**Comment by AffectedArc07** [comment]
I also threw a fix in for the `.gitignore` not ignoring the Midi2Piano build cache.

File: code/_globalvars/sensitive.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// They write as native BYOND globals, which means they cant be edited at all, and also use a format
+// Please only throw absolutely crucial do-not-view shit in here please. -aa07
+
+// MySQL configuration
+GLOBAL_REAL_VAR(sqladdress) = "localhost"
+GLOBAL_REAL_VAR(sqlport) = "3306"
+GLOBAL_REAL_VAR(sqlfdbkdb) = "test"
+GLOBAL_REAL_VAR(sqlfdbklogin) = "root"
+GLOBAL_REAL_VAR(sqlfdbkpass) = ""
+GLOBAL_REAL_VAR(sqlfdbktableprefix) = "erro_"
```
Comment (Crazylemon64) [inline code/_globalvars/sensitive.dm]: Alright, so this is where the db file went to - makes sense

File: code/_globalvars/unused.dm [hunk 2]
```diff
@@ -1 +0,0 @@
-var/going = 1.0
```
Comment (Crazylemon64) [inline code/_globalvars/unused.dm]: why did this even exist in the first place haha

File: code/controllers/subsystem/vote.dm [hunk 3]
```diff
@@ -164,14 +164,14 @@ SUBSYSTEM_DEF(vote)
...
 			if("gamemode")
-				if(master_mode != .)
+				if(GLOB.master_mode != .)
 					world.save_mode(.)
 					if(SSticker && SSticker.mode)
 						restart = 1
 					else
-						master_mode = .
-				if(!going)
-					going = 1
```
Comment (Crazylemon64) [inline code/controllers/subsystem/vote.dm]: oh, that's what "going" stood for. why was that file labeled "unused". why would anyone include something that is not used.

**Pull request merged**